packages/protocol/tipmanager/tipmanager.go

```diff
@@ -220,6 +214,84 @@ type markerPreviousBlockPair struct {
 	PreviousBlock *booker.Block
 }
 
+func (t *TipManager) addTip(block *scheduler.Block) (added bool) {
+	if !t.tips.Has(block) {
+		t.tips.Set(block, block)
+		// t.tipsConflictTracker.AddTip(block)
+		t.Events.TipAdded.Trigger(block)
+		return true
+	}
+
+	return false
+}
+
+// checkMonotonicity returns true if the block has any accepted or scheduled child.
+func (t *TipManager) checkMonotonicity(block *scheduler.Block) (anyScheduledOrAccepted bool) {
+	for _, child := range block.Children() {
+		if child.IsOrphaned() {
+			continue
+		}
+
+		if t.blockAcceptanceGadget.IsBlockAccepted(child.ID()) {
+			return true
+		}
+
+		if childBlock, exists := t.schedulerBlockRetrieverFunc(child.ID()); exists {
+			if childBlock.IsScheduled() {
+				return true
+			}
+		}
+	}
+
+	return false
+}
+
+// isStaleCommitment returns true if the block belongs to a commitment that is too far back in the past.
+func (t *TipManager) isStaleCommitment(block *scheduler.Block) (isOld bool) {
+	return block.Commitment().Index() < t.engine.Storage.Settings.LatestCommitment().Index()-1
```

Comment:
Isn't `t.engine.Storage.Settings.LatestCommitment().Index()` 0 at the beginning?